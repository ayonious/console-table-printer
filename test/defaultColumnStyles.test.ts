import { Table } from '../index';

describe('Testing defaultColumnStyles in Table', () => {
  it('should apply default styles to columns when not specified', () => {
    const defaultStyles = { alignment: 'center', color: 'blue', maxLen: 50 };
    const p = new Table({
      columns: [
        { name: 'col1' },
        { name: 'col2', alignment: 'left' },
      ],
      defaultColumnStyles: defaultStyles,
    });

    p.addRow({ col1: 'value1', col2: 'value2' });
    const rendered = p.render();
    expect(rendered).toContain('value1');
    expect(rendered).toContain('value2');
    // Check if default styles are applied
    expect(p.table.columns[0].alignment).toBe('center');
    expect(p.table.columns[0].color).toBe('blue');
    expect(p.table.columns[0].maxLen).toBe(50);
    // Check if specified styles override defaults
    expect(p.table.columns[1].alignment).toBe('left');

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
