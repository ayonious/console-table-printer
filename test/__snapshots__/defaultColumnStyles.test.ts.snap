// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing defaultColumnOptions in Table should apply default styles to columns when not specified 1`] = `
"┌────────────────────────────────────────────────────┬────────────────────────────────────────────────────┐
│[37m [0m[01m                       col1                       [0m │[37m [0m[01mcol2                                              [0m │
├────────────────────────────────────────────────────┼────────────────────────────────────────────────────┤
│[37m [0m[34m                      value1                      [0m │[37m [0m[34mvalue2                                            [0m │
└────────────────────────────────────────────────────┴────────────────────────────────────────────────────┘"
`;
